aio_kavenegar/client.py

```diff
@@ -0,0 +1,182 @@
+import json
+
+from typing import Literal
+
+import httpx
+
+from aio_kavenegar.exceptions import APIException, HTTPException
+from aio_kavenegar.types import KavenegarResponse
+
+
+# Default requests timeout in seconds.
+DEFAULT_TIMEOUT: int = 10
+DEFAULT_HOST: str = "api.kavenegar.com"
+DEFAULT_HEADERS: dict = {
+    "Accept": "application/json",
+    "Content-Type": "application/x-www-form-urlencoded",
+    "charset": "utf-8",
+}
+
+
+class AIOKavenegarAPI:
+    """
+    https://kavenegar.com/rest.html
+    """
+
+    def __init__(
+        self,
+        apikey: str,
+        timeout: int = DEFAULT_TIMEOUT,
+        host: str = DEFAULT_HOST,
+        headers: dict = DEFAULT_HEADERS,
+        proxies: dict | None = None,
+        secure: bool = True,
+    ) -> None:
+        """
+        :param str apikey: Kavengera API Key
+        :param int timeout: request timeout, default is 10
+        :param str host: Kavenegar API host, default is `api.kavenegar.com`
+        :param dict headers: headers used when requesting Kavenegar resources, default:
+            {
+                "Accept": "application/json",
+                "Content-Type": "application/x-www-form-urlencoded",
+                "charset": "utf-8",
+            }
+        :param dict proxies: Dictionary mapping protocol to the URL of the proxy:
+            {
+                'http': 'http://192.168.1.10:3128',
+                'https': 'http://192.168.1.10:3129',
+            }
```

Comment:
Why not get a proxy string and pass it to the client? For example, httpx can use socks proxies when the optional `socks` dependency is installed. This behavior unnecessarily limits the user's choices.

Reply:
@Alirezaja1384, Original library uses this schema and it makes migrations easy